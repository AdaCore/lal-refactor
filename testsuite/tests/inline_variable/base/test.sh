cp ./main.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 4 -SC 5
echo "--  -SL 4 -SC 5"
cat ./main.adb

cp ./main.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 5 -SC 6
echo "--  -SL 5 -SC 6"
cat ./main.adb

cp ./main.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 6 -SC 6
echo "--  -SL 6 -SC 6"
cat ./main.adb

cp ./main1.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 4 -SC 5
echo "--  main1 -SL 4 -SC 5"
cat ./main.adb

cp ./main1.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 4 -SC 11
echo "--  main1 -SL 4 -SC 11"
cat ./main.adb

cp ./main1.orig ./main.adb
inline_variable -P default.gpr -S main.adb -SL 4 -SC 17
echo "--  main1 -SL 4 -SC 17"
cat ./main.adb
