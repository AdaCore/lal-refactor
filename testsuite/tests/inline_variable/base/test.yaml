title: inline_variable

description: inline_variable tool test

driver: shell_script
