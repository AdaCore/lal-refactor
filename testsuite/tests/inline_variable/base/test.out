--  -SL 4 -SC 5
with Ada.Text_IO;

procedure Main is
   Tmp2 : Integer;
   Tmp3 : Integer := 1 + 1;
begin
   if (2 + 2) / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp2 / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   Tmp3 := Tmp3 + 2;
   if Tmp3 / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
--  -SL 5 -SC 6
with Ada.Text_IO;

procedure Main is
   Tmp  : Integer := 2 + 2;
   Tmp2 : Integer;
   Tmp3 : Integer := 1 + 1;
begin
   if Tmp / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp2 / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   Tmp3 := Tmp3 + 2;
   if Tmp3 / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
--  -SL 6 -SC 6
with Ada.Text_IO;

procedure Main is
   Tmp  : Integer := 2 + 2;
   Tmp2 : Integer;
   Tmp3 : Integer := 1 + 1;
begin
   if Tmp / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp2 / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   Tmp3 := Tmp3 + 2;
   if Tmp3 / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
--  main1 -SL 4 -SC 5
with Ada.Text_IO;

procedure Main is
   Tmp2, Tmp3 : Integer := 2 + 2;
begin
   if (2 + 2) / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp2 / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp3 / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
--  main1 -SL 4 -SC 11
with Ada.Text_IO;

procedure Main is
   Tmp, Tmp3 : Integer := 2 + 2;
begin
   if Tmp / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if (2 + 2) / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp3 / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
--  main1 -SL 4 -SC 17
with Ada.Text_IO;

procedure Main is
   Tmp, Tmp2 : Integer := 2 + 2;
begin
   if Tmp / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if Tmp2 / 2 = 1 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;

   if (2 + 2) / 2 = 2 then
      Ada.Text_IO.Put_Line ("Yes");
   end if;
end Main;
